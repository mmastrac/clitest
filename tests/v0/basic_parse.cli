#!/usr/bin/env clitest --v0

$ echo "a\nb\nc\n"
!!!
a
b
c

!!!
$ echo "a\nb\nc\n"
! a
! b
! c
!

$ echo "a
b
c
"
!!!
a
b
c

!!!

$ echo a\
b\
c
!!!
abc
!!!

$ echo "a\nb\nc"
%SET ENV_VAR
*

$ echo "$ENV_VAR"
!!!
a
b
c
!!!

$ VAR2="a\nb\nc" sh -c 'echo "$VAR2"'
!!!
a
b
c
!!!

$ echo "%{}"
! %{}
