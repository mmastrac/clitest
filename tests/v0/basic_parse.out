Running <root>/basic_parse.cli ...
printf "a\nb\nc\n\n"
---
a
b
c

---
✅ OK

printf "a\nb\nc\n\n"
---
a
b
c

---
✅ OK

echo "a
b
c
"
---
a
b
c

---
✅ OK

echo a\
b\
c
---
abc
---
✅ OK

printf "a\nb\nc\n"
---
a
b
c
---
✅ OK

printf "$ENV_VAR\n"
---
a
b
c
---
✅ OK

VAR2="a\nb\nc" sh -c 'printf "$VAR2\n"'
---
a
b
c
---
✅ OK

printf "%%{}\n"
---
%{}
---
✅ OK

<root>/basic_parse.cli PASSED
